tests/test_other.py

```diff
@@ -7596,6 +7596,57 @@ def test_binaryen_ignore_implicit_traps(self):
       print('sizes:', sizes)
       assert sizes[1] < sizes[0], 'ignoring implicit traps must reduce code size'
 
+  def test_binaryen_methods(self):
+    for method_init in ['interpret-asm2wasm', 'interpret-s-expr', 'asmjs', 'interpret-binary', 'asmjs,interpret-binary', 'interpret-binary,asmjs']:
+      # check success and failure for simple modes, only success for combined/fallback ones
+      for success in [1, 0] if ',' not in method_init else [1]:
+        method = method_init
+        command = [PYTHON, EMCC, '-o', 'a.wasm.js', '-s', 'BINARYEN=1', path_from_root('tests', 'hello_world.c')]
+        command += ['-s', 'BINARYEN_METHOD="' + method + '"']
+        print([method, ' : ', ' '.join(command), ' => ', success])
+        subprocess.check_call(command)
+
+        see_polyfill =  'var WasmJS = ' in open('a.wasm.js').read()
+
+        if method and 'interpret' not in method:
+          assert not see_polyfill, 'verify polyfill was not added - we specified a method, and it does not need it'
+        else:
+          assert see_polyfill, 'we need the polyfill'
+
+        def break_cashew():
+          asm = open('a.wasm.asm.js').read()
+          asm = asm.replace('"almost asm"', '"use asm"; var not_in_asm = [].length + (true || { x: 5 }.x);')
+          asm = asm.replace("'almost asm'", '"use asm"; var not_in_asm = [].length + (true || { x: 5 }.x);')
+          with open('a.wasm.asm.js', 'w') as o: o.write(asm)
+        if method.startswith('interpret-asm2wasm'):
+          try_delete('a.wasm.wast') # we should not need the .wast
+          if not success:
+            break_cashew() # we need cashew
+        elif method.startswith('interpret-s-expr'):
+          try_delete('a.wasm.asm.js') # we should not need the .asm.js
+          if not success:
+            try_delete('a.wasm.wast')
+        elif method.startswith('asmjs'):
+          try_delete('a.wasm.wast') # we should not need the .wast
+          break_cashew() # we don't use cashew, so ok to break it
+          if not success:
+            try_delete('a.wasm.js')
+        elif method.startswith('interpret-binary'):
+          try_delete('a.wasm.wast') # we should not need the .wast
+          try_delete('a.wasm.asm.js') # we should not need the .asm.js
+          if not success:
+            try_delete('a.wasm.wasm')
+        else:
+          1/0
```

Comment:
Now that we have future integer division, you can do `1//0` :D

Reply:
Ok ok, I wrote a proper throw ;)